Assets/_BForBoss/_Utility/Scripts/Editor/LevelDesignFeedback/ImageEditorWindow.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class ImageEditorWindow : EditorWindow
+    {
+        private const float TOOLBAR_HEIGHT = 30f;
+    
+        public Action<Texture2D> OnWindowClosed;
+        
+        private Texture2D _originalScreenShot = null;
+        private Texture2D _editedScreenShot = null;
+        private ImageEditorWindow _window;
+        
+        private bool _shouldCloseWindow = false;
+        private bool _drawThisFrame = false;
+        private bool _forceRepaint = false;
+
+        private Rect _screenShotRect;
+        private Color32 _brushColor = Color.black;
+        private int _currentBrushIndex = 1;
+        private int _brushSize;
+        private readonly string[] _brushSelections = new string[3] {"Small", "Medium", "Large"};
+        private readonly int[] _brushSizes = new int[3] {4, 7, 10};
+
+        public void OpenWindow(Texture2D screenShot)
+        {
+            if (screenShot == null)
+            {
+                return;
+            }
+            
+            _originalScreenShot = screenShot;
+            _editedScreenShot = CreateTextureCopy(_originalScreenShot);
+
+            Undo.undoRedoPerformed += OnUndoBrushStroke;
+            
+            _window = GetWindow<ImageEditorWindow>();
+            _window.minSize = new Vector2(screenShot.width, screenShot.height + TOOLBAR_HEIGHT);
+            _window.maxSize = _window.minSize;
+            _window.titleContent = new GUIContent("Edit ScreenShot");
+
+            _window.wantsMouseMove = true;
+            _window.wantsMouseEnterLeaveWindow = true;
+
+            _screenShotRect = new Rect(0,0,screenShot.width, screenShot.height);
+
+            _window.ShowPopup();
+        }
+
+        private void OnGUI()
+        {
+            if (_shouldCloseWindow)
+            {
+                Close();
+            }
+            
+            DrawScreenShotToEdit();
+            EditorGUILayout.Space(_editedScreenShot.height + 5f);
+            DrawToolbar();
+        }
+
+        private void DrawScreenShotToEdit()
+        {
+            EditorGUI.DrawPreviewTexture(_screenShotRect, _editedScreenShot);
+
+            if (mouseOverWindow != this)
+            {
+                return;
+            }
+            
+            Event evt = Event.current;
+            
+            if (evt.isMouse && evt.button == 0)
```

Comment:
Can you separate this line 74 - 98 into a separate method, so at least it's clear this is not actually drawing, but it's related to input